tests/infra_test.go

```diff
@@ -252,24 +253,26 @@ var _ = Describe("[Serial]Infrastructure", func() {
 				Expect(err).ShouldNot(HaveOccurred(), "failed listing kubevirt pods")
 				Expect(len(pods.Items)).To(BeNumerically(">", 0), "no kubevirt pods found")
 
-				By("finding all schedulable nodes")
-				schedulableNodesList := tests.GetAllSchedulableNodes(virtClient)
-				schedulableNodes := map[string]*k8sv1.Node{}
-				for _, node := range schedulableNodesList.Items {
-					schedulableNodes[node.Name] = node.DeepCopy()
-				}
-
 				By("selecting one compute only node that runs kubevirt components")
 				// master nodes should never have the CriticalAddonsOnly taint because core components might not
 				// tolerate this taint because it is meant to be used on compute nodes only. If we set this taint
 				// on a master node, we risk in breaking the test cluster.
-				for _, pod := range pods.Items {
-					node := schedulableNodes[pod.Spec.NodeName]
-					if _, isMaster := node.Labels["node-role.kubernetes.io/master"]; isMaster {
-						continue
+				Eventually(func() *k8sv1.Node {
+					schedulableNodesList := tests.GetAllSchedulableNodes(virtClient)
+					for _, node := range schedulableNodesList.Items {
```

Comment:
Can you name `node` as `schedulableNode` so the loop code is more clear.

Reply:
Done